internal/pkg/template/templates/task/cf.yml

```diff
@@ -89,6 +100,22 @@ Resources:
             Action: 'sts:AssumeRole'
       ManagedPolicyArns:
         - !Sub 'arn:${AWS::Partition}:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
+      {{- if .Secrets}}
+      Policies:
+        - PolicyName: 'PullSecrets'
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: 'Allow'
+                Action:
+                - ssm:GetParameters  
+                Condition:
+                  StringEquals:
+                    "ssm:ResourceTag/copilot-application": !If [ HasAppAndEnv, !Ref App, !Ref "AWS::NoValue" ]
+                    "ssm:ResourceTag/copilot-environment": !If [ HasAppAndEnv, !Ref Env, !Ref "AWS::NoValue" ]
+                    "ssm:ResourceTag/copilot-task": !If [ HasAppAndEnv, !Ref "AWS::NoValue", !Ref TaskName ]
```

Comment:
I really like the changes with the allowing access to the `copilot-application` and `copilot-environment` tags  -- makes total sense to me. I am not sure though how obvious it's going to be to use not using a copilot managed cluster that they need the `copilot-task` tag.

What if we had an experience like this?

```
$ copilot task run --default --secrets hello=/hello,world=arn:aws:secretsmanager:us-west-2:111122223333:secret:aes128-1a2b3c
Looks like you're requesting ssm:GetParameters to the following SSM parameters:
* /hello 
and secretsmanager:GetSecretValue to the following Secrets Manager secrets:
* arn:aws:secretsmanager:us-west-2:111122223333:secret:aes128-1a2b3c

Do you grant permission to the ECS/Fargate agent for these secrets? (y/N)
```

Or if they'd like to skip the prompt they can pass a flag similar to what the CDK does with `--require-approval never`

```
$ copilot task run --default --secrets hello=/hello --require-approval never
```

Then we will augment the Task Execution role such that only the input resources are granted access.

Let me know what you think!

Reply:
> Do you grant permission to the ECS/Fargate agent for these secrets? (y/N)

If they choose "N", do we just error out saying they can't use these secrets without granting access? And what are available options for `--require-approval`?

Reply:
> If they choose "N", do we just error out saying they can't use these secrets without granting access? 

I assume we'll abort then, but we can look to what the CDK does as well.

> And what are available options for --require-approval?

hmm good point for us none, maybe we can flip the flag `--acknowledge-iam`